go-controller/pkg/node/base_node_network_controller_dpu.go

```diff
@@ -158,53 +232,74 @@ func (bnnc *BaseNodeNetworkController) updatePodDPUConnStatusWithRetry(origPod *
 	if err != nil {
-		return fmt.Errorf("failed to get dpu annotation. %v", err)
+		klog.Infof("Failed to get VF representor for %s dpuConnDetail %+v: %v", podDesc, dpuCD, err)
+		return err
 	}
 
+	// set VfNetdevName so OVS interface can be added with external_ids:vf-netdev-name, and is able to
+	// be part of healthcheck.
+	ifInfo.VfNetdevName = vfRepName
+	klog.Infof("Adding VF representor %s for %s", vfRepName, podDesc)
 	err = cni.ConfigureOVS(context.TODO(), pod.Namespace, pod.Name, vfRepName, ifInfo, dpuCD.SandboxId, getter)
 	if err != nil {
 		// Note(adrianc): we are lenient with cleanup in this method as pod is going to be retried anyway.
-		_ = bnnc.delRepPort(vfRepName)
+		_ = bnnc.delRepPort(dpuCD, vfRepName, nadName, podDesc)
 		return err
 	}
 	klog.Infof("Port %s added to bridge br-int", vfRepName)
 
 	link, err := util.GetNetLinkOps().LinkByName(vfRepName)
 	if err != nil {
-		_ = bnnc.delRepPort(vfRepName)
+		_ = bnnc.delRepPort(dpuCD, vfRepName, nadName, podDesc)
 		return fmt.Errorf("failed to get link device for interface %s", vfRepName)
 	}
 
 	if err = util.GetNetLinkOps().LinkSetMTU(link, ifInfo.MTU); err != nil {
-		_ = bnnc.delRepPort(vfRepName)
+		_ = bnnc.delRepPort(dpuCD, vfRepName, nadName, podDesc)
 		return fmt.Errorf("failed to setup representor port. failed to set MTU for interface %s", vfRepName)
 	}
 
 	if err = util.GetNetLinkOps().LinkSetUp(link); err != nil {
-		_ = bnnc.delRepPort(vfRepName)
+		_ = bnnc.delRepPort(dpuCD, vfRepName, nadName, podDesc)
 		return fmt.Errorf("failed to setup representor port. failed to set link up for interface %s", vfRepName)
 	}
 
 	// Update connection-status annotation
 	// TODO(adrianc): we should update Status in case of error as well
 	connStatus := util.DPUConnectionStatus{Status: util.DPUConnectionStatusReady, Reason: ""}
-	err = bnnc.updatePodDPUConnStatusWithRetry(pod, &connStatus)
+	err = bnnc.updatePodDPUConnStatusWithRetry(pod, &connStatus, nadName)
 	if err != nil {
 		_ = util.GetNetLinkOps().LinkSetDown(link)
-		_ = bnnc.delRepPort(vfRepName)
+		_ = bnnc.delRepPort(dpuCD, vfRepName, nadName, podDesc)
 		return fmt.Errorf("failed to setup representor port. failed to set pod annotations. %v", err)
 	}
 	return nil
 }
 
 // delRepPort delete the representor of the VF from the ovs bridge
-func (bnnc *BaseNodeNetworkController) delRepPort(vfRepName string) error {
+func (bnnc *BaseNodeNetworkController) delRepPort(dpuCD *util.DPUConnectionDetails, vfRepName, nadName, podDesc string) error {
 	//TODO(adrianc): handle: clearPodBandwidth(pr.SandboxID), pr.deletePodConntrack()
-	klog.Infof("Delete VF representor %s port", vfRepName)
+	klog.Infof("Delete VF representor %s for %s", vfRepName, podDesc)
+	ifExists, sandbox, expectedNADName, err := util.GetOVSPortPodInfo(vfRepName)
+	if err != nil {
+		return fmt.Errorf(err.Error())
+	}
+	if !ifExists {
+		klog.Infof("VF representor %s for %s is not an OVS interface, nothing to do", vfRepName, podDesc)
```

Comment:
What is the case for a Pod's VF Rep to be not an OVS interface? Isn't this an unexpected error?

Reply:
it means this VF Rep is already unconfigured from br-int bridge.

Reply:
Is this an other condition where the DPU ovnkube-node is restarted, while the state in the DPU-Host's ovnkube-node changed?

I am wondering how we can get to a point where the VF Rep was already unconfigured, but we get here.

Reply:
I don't think this is why we added this code. We added the code only for later sandbox/NAD check, but for completeness of the code, we need to consider the case when ifExist is false. In case that happened (don't know how), I personally don't think we should return an error (probably log a Warning) as it does not have anything functionally wrong.